lib/logstash/inputs/rabbitmq.rb

```diff
@@ -131,17 +139,87 @@ def consume!
         end
       end
 
+      public
       def stop
         super
         shutdown_consumer
         close_connection
       end
 
+      public
       def shutdown_consumer
         return unless @consumer
         @consumer.gracefully_shut_down
       end
 
+      private
+      def get_header_value(value)
+        # Two kinds of values require exceptional treatment:
+        #
+        # String values are instances of
+        # com.rabbitmq.client.impl.LongStringHelper.ByteArrayLongString
+        # and we don't want to propagate those.
+        #
+        # List values are java.util.ArrayList objects and we need to
+        # recurse into them to convert any nested strings values.
+        if value.respond_to? "toArray"
+          value.map{|item| get_header_value(item) }
+        elsif value.respond_to? "toString"
+          value.toString
+        else
+          value
+        end
+      end
+
+      private
+      def get_headers(metadata)
+        if not metadata.headers.nil?
```

Comment:
I apologize. We really need to update our style guide. Would you mind using the [higher precedence](http://stackoverflow.com/questions/27429711/can-someone-explain-not-vs-in-ruby) `!` instead of `not`? 

The lower precedence operators obviously work fine, but they can be a subtle source of bugs if someone writes `if not something && something else`.